streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -194,7 +198,9 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
                     // Now, we should go ahead and complete the close because a half-closed task is no good to anyone.
                     task.closeDirty();
                 } finally {
-                    taskProducers.remove(task.id());
+                    if (taskProducers.containsKey(task.id())) {
+                        taskProducers.remove(task.id()).close();
+                    }
```

Comment:
Similar to above: this issue was introduced in the `StreamsProducer` PR. We nee to close the producer when we remove it.

Reply:
Probably need to change after rebase